src/conversion.rs

```diff
@@ -34,156 +37,286 @@ impl fmt::Display for Error {
     }
 }
 
-fn create_geo_coordinate<T>(coord: &Coord<T>) -> geo_types::Coordinate<T>
+impl<T> TryFrom<Wkt<T>> for geo_types::Geometry<T>
 where
     T: num_traits::Float,
 {
-    geo_types::Coordinate {
-        x: T::from(coord.x).unwrap(),
-        y: T::from(coord.y).unwrap(),
+    type Error = Error;
+
+    fn try_from(mut wkt: Wkt<T>) -> Result<Self, Self::Error> {
+        if wkt.items.len() == 1 {
+            Self::try_from(wkt.items.pop().unwrap())
+        } else {
+            Geometry::GeometryCollection(GeometryCollection(wkt.items)).try_into()
+        }
     }
 }
 
-fn try_into_point<T>(point: &Point<T>) -> Result<geo_types::Geometry<T>, Error>
+impl<T> From<Coord<T>> for geo_types::Coordinate<T>
 where
     T: num_traits::Float,
 {
-    match point.0 {
-        Some(ref c) => {
-            let geo_point: geo_types::Point<T> = (c.x, c.y).into();
-            Ok(geo_point.into())
+    fn from(coord: Coord<T>) -> geo_types::Coordinate<T> {
+        Self {
+            x: coord.x,
+            y: coord.y,
         }
-        None => Err(Error::PointConversionError),
     }
 }
 
-pub fn try_into_geometry<T>(geometry: &Geometry<T>) -> Result<geo_types::Geometry<T>, Error>
+impl<T> TryFrom<Point<T>> for geo_types::Point<T>
 where
     T: num_traits::Float,
 {
-    match geometry {
-        Geometry::Point(point) => try_into_point(point),
-        Geometry::LineString(linestring) => Ok(linestring.into()),
-        Geometry::Polygon(polygon) => Ok(polygon.into()),
-        Geometry::MultiLineString(multilinestring) => Ok(multilinestring.into()),
-        Geometry::MultiPoint(multipoint) => Ok(multipoint.into()),
-        Geometry::MultiPolygon(multipolygon) => Ok(multipolygon.into()),
-        Geometry::GeometryCollection(geometrycollection) => {
-            try_into_geometry_collection(geometrycollection)
+    type Error = Error;
+
+    fn try_from(point: Point<T>) -> Result<Self, Self::Error> {
+        match point.0 {
+            Some(coord) => Ok(Self::new(coord.x, coord.y)),
+            None => Err(Error::PointConversionError),
         }
     }
 }
 
-impl<'a, T> Into<geo_types::Geometry<T>> for &'a LineString<T>
+#[deprecated(since = "0.9", note = "use `geometry.try_into()` instead")]
+pub fn try_into_geometry<T>(geometry: &Geometry<T>) -> Result<geo_types::Geometry<T>, Error>
+where
+    T: num_traits::Float,
+{
+    geometry.clone().try_into()
+}
+
+impl<'a, T> From<&'a LineString<T>> for geo_types::Geometry<T>
```

Comment:
e.g. this extends the API offered by the previous `impl<'a, T> Into<geo_types::Geometry<T>> for &'a LineString<T>`